haystack/preview/components/file_converters/pypdf.py

```diff
@@ -0,0 +1,87 @@
+import logging
+from typing import List, Optional, Dict, Any, Union
+from pathlib import Path
+from tqdm import tqdm
+
+from haystack.preview.lazy_imports import LazyImport
+from haystack.preview import Document, component, default_to_dict, default_from_dict
+
+with LazyImport("Run 'pip install pypdf'") as pypdf:
+    from pypdf import PdfReader
+
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class PyPDFToDocument:
+    """
+    A component for converting a PDF file to a Document.
+    """
+
+    def __init__(self, id_hash_keys: Optional[List[str]] = None, progress_bar: bool = True):
+        """
+        Create a PyPDFToDocument component.
+
+        :param id_hash_keys: Generate the Document ID from a custom list of strings that refer to the Document's
+            attributes. Default: `None`
+        :param progress_bar: Whether to show a progress bar for the conversion process. Default: `True`
+        """
+        self.id_hash_keys = id_hash_keys or []
+        self.progress_bar = progress_bar
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serialize this component to a dictionary.
+        """
+        return default_to_dict(self, id_hash_keys=self.id_hash_keys, progress_bar=self.progress_bar)
```

Comment:
Why serializing the progress bar?

Reply:
It's a bool whether to use progress bars or not. Don't know actually what's the current recommended approach for progress bar usage in 2.0, will ask

Reply:
Ok, the word is that we are removing them and will use them if needed. Will remove now